### DISCUSSION TITLE
REACT Intermediary Data

### DISCUSSION THREAD

**Discussion by fimoefive** [nss-evening-cohort-14/evening-client-side#234] [Help Me!] [answered]
Casey, Honey-Rae, Martin

Have you searched the other help tickets?
Did not see any recent issue tickets with similar problems as this current issue.

Describe your issue
We are having an issue we are getting arrays instead of objects. It is an array attached into an object for some reason.

When we are getting the intermediary data from firebase we are getting an array attached to an object

List any error messages that appear

What have you googled?
https://www.freecodecamp.org/news/javascript-es6-promises-for-beginners-resolve-reject-and-chaining-explained/

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

What have you tried? Tell us everything you have tried.

Aja has helped us with a nest function with .map of a Join table, then .filter channel, then a .filter user function. 

Repo (not sure what file to show you for this error)
https://github.com/nss-evening-cohort-14/awesome-sauce-react-slack-simulation/blob/hr-intermediaryTables/src/helpers/data/userChannelData.js#L19

**Comment by HunterJuneau** [comment]
Could you copy and paste the data you are getting?

**Reply by caseywalker** [reply]
The problem is that we are not getting back anything. 

We are taking the approach to intermediary data that we took in the last group project prior to React.

The issue is that we are trying to append an array of data from firebase to another object, and might not be the best way to retrieve the data from firebase. We are essentially just trying to get all users associated with a channel for Slack Simulation.

**Comment by TrinityChristiana** [comment, marked as answer]
Looks like you're using `.firebaseKey` on things that [should be using `.channelKey`](https://github.com/nss-evening-cohort-14/awesome-sauce-react-slack-simulation/blob/hr-intermediaryTables/src/helpers/data/userChannelData.js#L23) or [`.userKey`](https://github.com/nss-evening-cohort-14/awesome-sauce-react-slack-simulation/blob/hr-intermediaryTables/src/helpers/data/userChannelData.js#L25). Make sure you're using the right keys from the join tables to relate to the other tables.

**Reply by fimoefive** [reply]
Thank you, for taking the time to explain in class today the different ways such as a drop down in the Channel Form or a useParams hook to join Organization and Channels together. It was very insightful

**Comment by drteresavasquez** [comment]
@fimoefive @caseywalker  did this resolve your issue? Please follow up.

**Reply by caseywalker** [reply]
Apologies, I was out of pocket this weekend. 

Tonight I plan to refactor our intermediary calls and see if we can get it to return what we need.

**Reply by drteresavasquez** [reply]
No worries. Is this ticket resolved then? Or are you going to wait until implementing and see where you get with it?

**Reply by caseywalker** [reply]
Currently refactoring, if I get the data back that we need, it will be resolved. Should be less than an hour.
